types/wlr_virtual_keyboard_v1.c

```diff
@@ -99,6 +99,7 @@ static void virtual_keyboard_destroy_resource(struct wl_resource *resource) {
 		virtual_keyboard_from_resource(resource);
 	wlr_signal_emit_safe(&keyboard->events.destroy, keyboard);
 	wl_list_remove(&keyboard->link);
+	wlr_seat_set_keyboard(keyboard->seat, NULL);
```

Comment:
We should only do this if `keyboard->seat->keyboard_state.keyboard == keyboard->input_device->keyboard` (ie. if the active keyboard on the seat is the virtual keyboard).

Reply:
Oh, a better fix would be to listen to the keyboard `destroy` signal in the seat.